### PR TITLE
First pull request for Enumerable Numbers project

### PR DESCRIPTION
**Pull request by bruna-genz**

### CODE REVIEW

File: enumerable_number.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  def my_count
+    result = length
+    unless block_given?
+      my_each do |element|
+        result += 1 if yield(element)
+      end
+    end
+    result
+  end
```
Comment (maelfosso) [inline enumerable_number.rb:84]: Sorry :disappointed:  there are some scenario not implemented
```ruby
rb(main):029:0> ary.count 
=> 4
irb(main):030:0> ary.my_count 
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):30
        3: from enumerable_number.rb:73:in `my_count'
        2: from enumerable_number.rb:7:in `my_each'
        1: from enumerable_number.rb:74:in `block in my_count'
LocalJumpError (no block given (yield))
```
Same thing here
```ruby
irb(main):031:0> ary.count(2)
=> 2
irb(main):032:0> ary.my_count(2)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):32
        1: from enumerable_number.rb:70:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

Error when counting, now
```irb(main):033:0> ary.count{ |x| x%2==0 }
=> 3
irb(main):034:0> ary.my_count{ |x| x%2==0 }
=> 4
```

File: enumerable_number.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    else
+      result = arg1
+      operation = arg2.to_proc
+      my_each do |element|
+        result = operation.call(result, element)
+      end
+    end
+    result
+  end
+end
```
Comment (maelfosso) [inline enumerable_number.rb:112]: Sorry there are errors
```ruby
irb(main):037:0> [1, 2, 4, 2].inject { |sum, n| sum + n } 
=> 9
irb(main):038:0> [1, 2, 4, 2].my_inject { |sum, n| sum + n } 
=> 10
irb(main):040:0> [1, 2, 4, 2].inject(2) { |sum, n| sum + n } 
=> 11
irb(main):041:0> [1, 2, 4, 2].my_inject(2) { |sum, n| sum + n } 
=> 10
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
HEllo @bruna-genz ,

# :clap: Nice job :smile: 

Please take a look below to do the needed changes.
Kindly make those changes and submit a new code review request.

Thank

File: enumerable_number.rb (hunk 2 above)
Comment (maelfosso) [reply enumerable_number.rb:112]: You fixed those bugs :smile: 
But there is a new bugs
```ruby
irb(main):007:0> [1, 2, 4, 2].inject { |prod, n| prod * n }
=> 16
irb(main):008:0> [1, 2, 4, 2].my_inject { |prod, n| prod * n }
=> 0
```

File: enumerable_number.rb (hunk 1 above)
Comment (maelfosso) [reply enumerable_number.rb:84]: Great :+1: 
:smile:

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello @bruna-genz 

Really nice improvement :smile: 
But there is just one bug to fix.
Take a look at the `my_inject` method.

Kindly make the changes and submit a new code review request, please.

File: enumerable_number.rb [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    result = []
+    if proc
+      my_each { |element| result << proc.call(element) }
+    else
+      my_each { |element| result << yield(element) }
+    end
+    result
+  end
+
+  def my_inject(arg1 = nil, arg2 = nil)
```
Comment (samgaco) [inline enumerable_number.rb:96]: working now :100:

**Review by samgaco — APPROVED** [review]
Hi @bruna-genz :high_brightness: 

The project is approved :white_check_mark: , you may merge to the master branch and submit the project completion form. Regards :hammer: 

------------------------

* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

**Pull request merged**